mgr/src/server/datastore/nodes.cr

```diff
@@ -6,90 +6,67 @@ require "db"
 module Datastore
   struct NodeView
     include DB::Serializable
-    property id = "", name = "", endpoint = "", pool_id = "", pool_name = ""
+    property id = "", name = "", endpoint = ""
   end
 
   private def nodes_query
     "SELECT nodes.id AS id,
             nodes.name AS name,
-            nodes.endpoint AS endpoint,
-            pools.id AS pool_id,
-            pools.name AS pool_name
+            nodes.endpoint AS endpoint
      FROM nodes
-     INNER JOIN pools ON nodes.pool_id = pools.id
     "
   end
 
   private def nodes_query_order_by
-    " ORDER BY pools.created_on DESC, nodes.created_on DESC "
-  end
-
-  private def grouped_nodes(nodes)
-    nodes.map do |row|
-      node = MoanaTypes::Node.new
-      node.id = row.id
-      node.name = row.name
-      node.endpoint = row.endpoint
-      node.pool = MoanaTypes::Pool.new
-      node.pool.id = row.pool_id
-      node.pool.name = row.pool_name
-
-      node
-    end
+    " ORDER BY nodes.created_on DESC "
   end
 
   def list_nodes
-    grouped_nodes(connection.query_all(nodes_query + nodes_query_order_by, as: NodeView))
+    connection.query_all(nodes_query + nodes_query_order_by, as: MoanaTypes::Node)
   end
 
-  def list_nodes_by_user(user_id)
-    pool_ids = viewable_pool_ids(user_id)
-    return list_nodes if pool_ids.includes?("all")
+  # def list_nodes_by_user(user_id)
+  #   pool_ids = viewable_pool_ids(user_id)
+  #   return list_nodes if pool_ids.includes?("all")
 
-    query = nodes_query + " WHERE pools.id IN (#{(pool_ids.map { |_| "?" }).join(",")}) " + nodes_query_order_by
-    grouped_nodes(connection.query_all(query, args: pool_ids, as: NodeView))
-  end
-
-  def list_nodes(pool_name)
-    grouped_nodes(connection.query_all(nodes_query + " WHERE pools.name = ?" + nodes_query_order_by, pool_name, as: NodeView))
-  end
+  #   query = nodes_query + " WHERE pools.id IN (#{(pool_ids.map { |_| "?" }).join(",")}) " + nodes_query_order_by
+  #   connection.query_all(query, args: pool_ids, as: MoanaTypes::Node)
+  # end
 
-  def get_nodes(pool_name, node_names)
+  def get_nodes(node_names)
```

Comment:
Now a node can be part of different pools?

Reply:
Yes. If a node consists of two storage units, one can be used with one pool and the other one can be used by the second pool.